apps/site/src/app/schedule/components/ShiftingCountdown/ShiftingCountdown.tsx

```diff
@@ -0,0 +1,151 @@
+"use client";
+
+import { AnimatePresence, motion } from "framer-motion";
+import { useEffect, useRef, useState } from "react";
+import BarLoader from "../BarLoader";
+import clsx from "clsx";
+import { Fireworks } from "@fireworks-js/react";
+
+import styles from "./ShiftingCountdown.module.scss";
+
+// NOTE: Change this date to whatever date you want to countdown to :)
+const HACKING_STARTS = "2024-01-26T18:00:00.000";
+const HACKING_ENDS = "2024-01-28T24:00:00.000";
+// const HACKING_STARTS = "2023-12-06T22:23:00.000"; // Test when countdown ends
+// HACKING_STARTS.setSeconds(10);
+// const HACKING_ENDS = new Date();
+
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+const HOUR = MINUTE * 60;
+const DAY = HOUR * 24;
+
+const ShiftingCountdown = () => {
+	const intervalRef = useRef<ReturnType<typeof setInterval> | null>(null);
+	const [showTimer, setShowTimer] = useState(false);
+	const [showFireworks, setShowFireworks] = useState(false);
+	const [timerLabel, setTimerLabel] = useState("Hacking Starts");
+
+	const [remaining, setRemaining] = useState({
+		days: 0,
+		hours: 0,
+		minutes: 0,
+		seconds: 0,
+	});
+
+	useEffect(() => {
+		intervalRef.current = setInterval(handleCountdown, 1000);
+
+		return () => clearInterval(intervalRef.current || undefined);
```

Comment:
Question/suggestion: is there a particular reason the component needs to keep track of the interval ID? If not, we can manage it only within the scope of the setup function.

Reply:
Yea the interval can be managed only in the scope off the UseEffect callback, but I believe it is good practice to assign a reference to an interval id to persist past component re-renders. The component I believe will not re-render, so it technically is not necessary.

Reply:
Nothing else in the component needs to be aware of the interval ID, so there's no point in maintaining the ref. Aside: every state change will trigger a re-render, so I'm not sure what you mean by the second sentence.

Reply:
kk ill remove it